core/asset/service.go

```diff
@@ -44,7 +43,7 @@ type ServiceDeps struct {
 	Config        Config
 }
 
-func NewService(deps ServiceDeps) (service *Service, cancel func()) {
```

Comment:
please return the cancel func here. this is essential during shutdown process.

Reply:
there is no cancel func in the NewService function to return.
What should we return?

Reply:
exactly the point, please return the cancel function and its implementation for calling the cancel functions.